### PR TITLE
Add 2 methods to RoutingModel class

### PR DESCRIPTION
**Pull request by starchow**
Dear @ankane,

Thank you for making this gem.

When using it to solve Vehicle Routing Problem, I faced with 2 problems
1. Sometimes OR-tools can't solve the problem cause the parameters are not solvable. I usually get this error message `[BUG] Segmentation fault at 0x0000000000000018` when I call `solution.value(routing.next_var(index))`
So, I added RoutingModel status method to check it before call routing.next_var
https://google.github.io/or-tools/cpp_routing/classoperations__research_1_1_routing_model.html#a67a0db04d321a74b7e7fcfd3f1a3f70b

2. I also added this method add_dimension_with_vehicle_transits to solve Vehicle Routing with Different Vehicle Speed Problem 
https://google.github.io/or-tools/cpp_routing/classoperations__research_1_1_routing_model.html#a66b87fec514d648c751ad9d4df09d05b

Please review this pull request,
Thanks

### CODE REVIEW

File: ext/or-tools/ext.cpp [hunk 1]
```diff
@@ -816,6 +816,23 @@ void Init_ext()
       })
     .define_method("depot", &RoutingModel::GetDepot)
     .define_method("size", &RoutingModel::Size)
+    .define_method("status", *[](RoutingModel& self) {
+        auto status = self.status();
+
+        if (status == RoutingModel::ROUTING_NOT_SOLVED ) {
+          return Symbol("routing_not_solved");
```
Comment (ankane) [inline ext/or-tools/ext.cpp:823]: Let's remove the `routing_` prefix from the return values.

File: test/routing_test.rb [hunk 2]
```diff
@@ -726,6 +728,89 @@ def test_penalties
...
+    ]
+    
+    data[:vehicle_speeds] = [15, 15, 15, 15]
+    data[:num_vehicles] = 4
+    data[:depot] = 0
+
+    manager = ORTools::RoutingIndexManager.new(data[:distance_matrix].size, data[:num_vehicles], data[:depot])
+    routing = ORTools::RoutingModel.new(manager)
+
+    transit_callback_indecies = []
```
Comment (ankane) [inline test/routing_test.rb:760]: `indecies` -> `indices`

**Review by ankane — COMMENTED** [review]
Hey @starchow, thanks for the PR! Looks really great. I left two small comments inline.

**Comment by starchow** [comment]
> Hey @starchow, thanks for the PR! Looks really great. I left two small comments inline.

I've just updated as your comments, thanks for your reviewing

**Comment by ankane** [comment]
Awesome, thanks @starchow!

**Pull request merged**